### PR TITLE
Generate configs for tagging tests from template

### PR DESCRIPTION
**Pull request by gdavison**
### Description

Generates configs for tagging tests from template file, located in `testdata/tmpl`.

Currently does not support configs that:

* take parameters in addition to `rName`

### Output from Acceptance Testing

```console
% make testacc TESTS=tags PKG=acmpca

--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_nullOverlappingResourceTag (53.81s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_emptyResourceTag (56.70s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_nullNonOverlappingResourceTag (57.60s)
--- PASS: TestAccACMPCACertificateAuthority_tags_null (60.47s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_updateToResourceOnly (68.60s)
--- PASS: TestAccACMPCACertificateAuthority_tags_AddOnUpdate (69.38s)
--- PASS: TestAccACMPCACertificateAuthority_tags_EmptyTag_OnUpdate_Replace (69.67s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_updateToProviderOnly (69.69s)
--- PASS: TestAccACMPCACertificateAuthority_tags_EmptyTag_OnCreate (72.65s)
--- PASS: TestAccACMPCACertificateAuthority_tags_EmptyTag_OnUpdate_Add (82.69s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_nonOverlapping (89.21s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_overlapping (91.04s)
--- PASS: TestAccACMPCACertificateAuthority_tags_DefaultTags_providerOnly (107.98s)
--- PASS: TestAccACMPCACertificateAuthority_tags (111.05s)
```

```console
% make testacc TESTS=tags PKG=TestAccAccessAnalyzer_serial/Analyzer/tags

--- PASS: TestAccAccessAnalyzer_serial (666.65s)
    --- PASS: TestAccAccessAnalyzer_serial/Analyzer (666.65s)
        --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags (666.65s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/basic (33.70s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/null (12.38s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/AddOnUpdate (15.49s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/EmptyTag_OnCreate (17.60s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/EmptyTag_OnUpdate_Add (23.37s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/EmptyTag_OnUpdate_Replace (15.60s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_providerOnly (116.92s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_nonOverlapping (85.62s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_overlapping (109.05s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_updateToProviderOnly (54.24s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_updateToResourceOnly (33.46s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_emptyResourceTag (50.19s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_nullOverlappingResourceTag (49.82s)
            --- PASS: TestAccAccessAnalyzer_serial/Analyzer/tags/DefaultTags_nullNonOverlappingResourceTag (49.20s)
```

```console
% make testacc TESTS=tags PKG=batch

--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag (29.49s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag (32.10s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag (32.61s)
--- PASS: TestAccBatchSchedulingPolicy_tags_null (46.06s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly (54.95s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly (57.30s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_nullNonOverlappingResourceTag (41.58s)
--- PASS: TestAccBatchComputeEnvironment_tags_null (74.27s)
--- PASS: TestAccBatchComputeEnvironment_tags_AddOnUpdate (89.52s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_nullOverlappingResourceTag (45.50s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add (95.01s)
--- PASS: TestAccBatchComputeEnvironment_tags_EmptyTag_OnCreate (97.03s)
--- PASS: TestAccBatchComputeEnvironment_tags_EmptyTag_OnUpdate_Replace (97.06s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_emptyResourceTag (42.23s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace (68.18s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping (100.54s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_overlapping (100.75s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_updateToResourceOnly (67.53s)
--- PASS: TestAccBatchComputeEnvironment_tags_EmptyTag_OnUpdate_Add (126.58s)
--- PASS: TestAccBatchJobDefinition_tags (137.50s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_updateToProviderOnly (63.53s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_providerOnly (138.50s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_nonOverlapping (138.91s)
--- PASS: TestAccBatchSchedulingPolicy_tags_EmptyTag_OnUpdate_Replace (64.67s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_overlapping (108.78s)
--- PASS: TestAccBatchJobDefinition_tags_AddOnUpdate (63.26s)
--- PASS: TestAccBatchSchedulingPolicy_tags_EmptyTag_OnCreate (76.40s)
--- PASS: TestAccBatchComputeEnvironment_tags (175.46s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnCreate (76.60s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_providerOnly (178.93s)
--- PASS: TestAccBatchSchedulingPolicy_tags_AddOnUpdate (74.43s)
--- PASS: TestAccBatchJobDefinition_tags_null (65.82s)
--- PASS: TestAccBatchSchedulingPolicy_tags_EmptyTag_OnUpdate_Add (109.71s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_nonOverlapping (126.72s)
--- PASS: TestAccBatchJobQueue_tags_DefaultTags_nonOverlapping (218.25s)
--- PASS: TestAccBatchSchedulingPolicy_tags_DefaultTags_providerOnly (154.05s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_nullNonOverlappingResourceTag (53.58s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_updateToProviderOnly (95.50s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_updateToResourceOnly (88.90s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_nullOverlappingResourceTag (56.06s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_emptyResourceTag (56.19s)
--- PASS: TestAccBatchSchedulingPolicy_tags (130.52s)
--- PASS: TestAccBatchComputeEnvironment_tags_DefaultTags_overlapping (97.21s)
--- PASS: TestAccBatchJobQueue_tags_DefaultTags_emptyResourceTag (138.81s)
--- PASS: TestAccBatchJobQueue_tags_EmptyTag_OnUpdate_Replace (183.91s)
--- PASS: TestAccBatchJobQueue_tags_AddOnUpdate (168.30s)
--- PASS: TestAccBatchJobQueue_tags_DefaultTags_updateToProviderOnly (164.56s)
--- PASS: TestAccBatchJobQueue_tags_DefaultTags_updateToResourceOnly (166.87s)
--- PASS: TestAccBatchJobQueue_tags_EmptyTag_OnCreate (168.07s)
--- PASS: TestAccBatchJobQueue_tags_EmptyTag_OnUpdate_Add (210.27s)
--- PASS: TestAccBatchJobQueue_tags_DefaultTags_providerOnly (226.66s)
--- PASS: TestAccBatchJobQueue_tags_DefaultTags_overlapping (151.84s)
--- PASS: TestAccBatchJobQueue_tags (162.38s)
```

```console
% make testacc TESTS=tags PKG=iam

--- PASS: TestAccIAMUserDataSource_tags (77.29s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_emptyResourceTag (89.08s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_nullNonOverlappingResourceTag (89.79s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_nullOverlappingResourceTag (98.74s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_emptyResourceTag (99.72s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_nullNonOverlappingResourceTag (106.49s)
--- PASS: TestAccIAMUser_tags_null (110.63s)
--- PASS: TestAccIAMUser_tags_AddOnUpdate (134.35s)
--- PASS: TestAccIAMServiceLinkedRole_tags_EmptyTag_OnUpdate_Replace (144.21s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_updateToProviderOnly (144.70s)
--- PASS: TestAccIAMUser_tags_EmptyTag_OnCreate (148.04s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_updateToResourceOnly (153.36s)
--- PASS: TestAccIAMRole_tags_DefaultTags_nullNonOverlappingResourceTag (65.69s)
--- PASS: TestAccIAMServiceLinkedRole_tags_null (96.39s)
--- PASS: TestAccIAMServiceLinkedRole_tags_EmptyTag_OnUpdate_Add (190.08s)
--- PASS: TestAccIAMRole_tags_DefaultTags_nonOverlapping (194.02s)
--- PASS: TestAccIAMRole_tags_DefaultTags_nullOverlappingResourceTag (59.84s)
--- PASS: TestAccIAMServiceLinkedRole_tags_AddOnUpdate (106.77s)
--- PASS: TestAccIAMRole_tags_DefaultTags_emptyResourceTag (54.04s)
--- PASS: TestAccIAMRole_tags_DefaultTags_overlapping (198.76s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_nonOverlapping (199.85s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_overlapping (205.70s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_nullOverlappingResourceTag (52.49s)
--- PASS: TestAccIAMServiceLinkedRole_tags_EmptyTag_OnCreate (131.74s)
--- PASS: TestAccIAMRole_tags_DefaultTags_updateToResourceOnly (79.91s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_emptyResourceTag (50.51s)
--- PASS: TestAccIAMRole_tags_DefaultTags_updateToProviderOnly (78.81s)
--- PASS: TestAccIAMServerCertificate_tags (130.31s)
--- PASS: TestAccIAMSAMLProvider_tags (125.24s)
--- PASS: TestAccIAMUser_tags (234.55s)
--- PASS: TestAccIAMInstanceProfile_tags (235.11s)
--- PASS: TestAccIAMServiceLinkedRole_tags_DefaultTags_providerOnly (244.19s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_updateToResourceOnly (74.11s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_updateToProviderOnly (71.19s)
--- PASS: TestAccIAMVirtualMFADevice_tags_null (60.12s)
--- PASS: TestAccIAMUser_tags_DefaultTags_nullNonOverlappingResourceTag (41.66s)
--- PASS: TestAccIAMUser_tags_DefaultTags_nullOverlappingResourceTag (41.87s)
--- PASS: TestAccIAMVirtualMFADevice_tags_EmptyTag_OnUpdate_Replace (70.47s)
--- PASS: TestAccIAMUser_tags_DefaultTags_emptyResourceTag (44.54s)
--- PASS: TestAccIAMVirtualMFADevice_tags_AddOnUpdate (72.01s)
--- PASS: TestAccIAMVirtualMFADevice_tags_EmptyTag_OnCreate (83.64s)
--- PASS: TestAccIAMUser_tags_DefaultTags_updateToResourceOnly (65.17s)
--- PASS: TestAccIAMUser_tags_DefaultTags_updateToProviderOnly (71.14s)
--- PASS: TestAccIAMServiceLinkedRole_tags (209.75s)
--- PASS: TestAccIAMVirtualMFADevice_tags_EmptyTag_OnUpdate_Add (110.58s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_nullNonOverlappingResourceTag (43.71s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_overlapping (118.12s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_nonOverlapping (121.53s)
--- PASS: TestAccIAMUser_tags_EmptyTag_OnUpdate_Replace (72.10s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_EmptyTag_OnUpdate_Replace (69.53s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_updateToResourceOnly (67.60s)
--- PASS: TestAccIAMOpenidConnectProviderDataSource_tags (36.18s)
--- PASS: TestAccIAMUser_tags_DefaultTags_overlapping (113.67s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_AddOnUpdate (70.84s)
--- PASS: TestAccIAMVirtualMFADevice_tags_DefaultTags_providerOnly (152.80s)
--- PASS: TestAccIAMUser_tags_DefaultTags_nonOverlapping (113.63s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_EmptyTag_OnCreate (80.09s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_updateToProviderOnly (68.92s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_null (60.03s)
--- PASS: TestAccIAMVirtualMFADevice_tags (148.39s)
--- PASS: TestAccIAMUser_tags_EmptyTag_OnUpdate_Add (105.50s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_EmptyTag_OnUpdate_Add (105.35s)
--- PASS: TestAccIAMRoleDataSource_tags (36.06s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_updateToResourceOnly (72.49s)
--- PASS: TestAccIAMUser_tags_DefaultTags_providerOnly (149.96s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_nullNonOverlappingResourceTag (45.52s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_emptyResourceTag (42.24s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_nullOverlappingResourceTag (47.41s)
--- PASS: TestAccIAMRole_tags_null (61.36s)
--- PASS: TestAccIAMRole_tags_EmptyTag_OnUpdate_Replace (74.28s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_nullOverlappingResourceTag (47.86s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_emptyResourceTag (45.21s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_overlapping (120.35s)
--- PASS: TestAccIAMRole_tags_AddOnUpdate (74.97s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_nonOverlapping (121.85s)
--- PASS: TestAccIAMRole_tags_EmptyTag_OnCreate (84.65s)
--- PASS: TestAccIAMRole_tags_EmptyTag_OnUpdate_Add (113.04s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_updateToResourceOnly (77.14s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_updateToProviderOnly (73.81s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags (154.38s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_updateToProviderOnly (69.49s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_providerOnly (159.55s)
--- PASS: TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Replace (74.71s)
--- PASS: TestAccIAMInstanceProfile_tags_null (59.02s)
--- PASS: TestAccIAMRole_tags_DefaultTags_providerOnly (161.30s)
--- PASS: TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Replace (75.59s)
--- PASS: TestAccIAMInstanceProfile_tags_AddOnUpdate (71.69s)
--- PASS: TestAccIAMPolicy_tags_EmptyTag_OnUpdate_Add (108.55s)
--- PASS: TestAccIAMRole_tags (159.36s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_nullNonOverlappingResourceTag (42.61s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_providerOnly (156.79s)
--- PASS: TestAccIAMPolicy_tags_null (54.65s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_overlapping (122.50s)
--- PASS: TestAccIAMInstanceProfile_tags_DefaultTags_nonOverlapping (123.65s)
--- PASS: TestAccIAMOpenIDConnectProvider_tags_DefaultTags_nullOverlappingResourceTag (31.02s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_overlapping (111.88s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_nonOverlapping (112.02s)
--- PASS: TestAccIAMPolicy_tags_AddOnUpdate (58.93s)
--- PASS: TestAccIAMInstanceProfile_tags_EmptyTag_OnCreate (70.69s)
--- PASS: TestAccIAMInstanceProfile_tags_EmptyTag_OnUpdate_Add (99.53s)
--- PASS: TestAccIAMPolicy_tags_EmptyTag_OnCreate (63.38s)
--- PASS: TestAccIAMPolicy_tags_DefaultTags_providerOnly (113.48s)
--- PASS: TestAccIAMPolicy_tags (75.59s)
```

```console
% make testacc TESTS=tags PKG=servicecatalog

--- SKIP: TestAccServiceCatalogPortfolio_tags_EmptyTag_OnCreate (0.00s)
--- SKIP: TestAccServiceCatalogPortfolio_tags_EmptyTag_OnUpdate_Add (0.00s)
--- SKIP: TestAccServiceCatalogPortfolio_tags_EmptyTag_OnUpdate_Replace (0.00s)
--- SKIP: TestAccServiceCatalogPortfolio_tags_DefaultTags_emptyResourceTag (0.00s)
--- SKIP: TestAccServiceCatalogProduct_tags_EmptyTag_OnCreate (0.00s)
--- SKIP: TestAccServiceCatalogProduct_tags_EmptyTag_OnUpdate_Add (0.00s)
--- SKIP: TestAccServiceCatalogProduct_tags_EmptyTag_OnUpdate_Replace (0.00s)
--- SKIP: TestAccServiceCatalogProduct_tags_DefaultTags_emptyResourceTag (0.00s)
--- SKIP: TestAccServiceCatalogProvisionedProduct_tags_EmptyTag_OnCreate (0.00s)
--- SKIP: TestAccServiceCatalogProvisionedProduct_tags_EmptyTag_OnUpdate_Add (0.00s)
--- SKIP: TestAccServiceCatalogProvisionedProduct_tags_EmptyTag_OnUpdate_Replace (0.00s)
--- SKIP: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_emptyResourceTag (0.00s)
--- PASS: TestAccServiceCatalogProduct_tags_null (59.31s)
--- PASS: TestAccServiceCatalogProduct_tags_AddOnUpdate (96.28s)
--- PASS: TestAccServiceCatalogPortfolio_tags (111.79s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_nullNonOverlappingResourceTag (151.63s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_updateToProviderOnly (156.71s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_updateToResourceOnly (160.92s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_nullOverlappingResourceTag (162.84s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_updateToResourceOnly (166.22s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_null (170.18s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_nullNonOverlappingResourceTag (186.30s)
--- PASS: TestAccServiceCatalogProduct_tags (203.95s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_nullOverlappingResourceTag (210.70s)
--- PASS: TestAccServiceCatalogPortfolio_tags_AddOnUpdate (61.11s)
--- PASS: TestAccServiceCatalogPortfolio_tags_null (50.67s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_updateToProviderOnly (183.77s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_AddOnUpdate (244.58s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags (247.99s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_nonOverlapping (264.38s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_nonOverlapping (292.06s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_overlapping (296.11s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_updateToResourceOnly (202.32s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_nullNonOverlappingResourceTag (191.33s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_overlapping (345.80s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_nullOverlappingResourceTag (194.82s)
--- PASS: TestAccServiceCatalogProduct_tags_DefaultTags_providerOnly (365.49s)
--- PASS: TestAccServiceCatalogPortfolio_tags_DefaultTags_providerOnly (202.13s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_updateToProviderOnly (219.45s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_providerOnly (384.55s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_nonOverlapping (293.58s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags_DefaultTags_overlapping (396.85s)
```

```console
% make testacc TESTS=tags PKG=xray

--- PASS: TestAccXRayGroup_tags_DefaultTags_nullNonOverlappingResourceTag (47.63s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_emptyResourceTag (47.77s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_nullOverlappingResourceTag (47.84s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_nullNonOverlappingResourceTag (49.84s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_nullOverlappingResourceTag (49.99s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_updateToResourceOnly (73.71s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_updateToResourceOnly (73.93s)
--- PASS: TestAccXRaySamplingRule_tags_AddOnUpdate (75.58s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_updateToProviderOnly (76.13s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_updateToProviderOnly (76.26s)
--- PASS: TestAccXRaySamplingRule_tags_EmptyTag_OnUpdate_Replace (76.74s)
--- PASS: TestAccXRaySamplingRule_tags_EmptyTag_OnCreate (91.92s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_emptyResourceTag (52.76s)
--- PASS: TestAccXRaySamplingRule_tags_null (61.64s)
--- PASS: TestAccXRaySamplingRule_tags_EmptyTag_OnUpdate_Add (114.42s)
--- PASS: TestAccXRayGroup_tags_EmptyTag_OnUpdate_Add (114.67s)
--- PASS: TestAccXRayGroup_tags_EmptyTag_OnUpdate_Replace (69.32s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_nonOverlapping (122.52s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_overlapping (123.27s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_nonOverlapping (125.14s)
--- PASS: TestAccXRayGroup_tags_null (52.43s)
--- PASS: TestAccXRayGroup_tags_AddOnUpdate (60.61s)
--- PASS: TestAccXRayGroup_tags_EmptyTag_OnCreate (66.60s)
--- PASS: TestAccXRaySamplingRule_tags_DefaultTags_providerOnly (141.56s)
--- PASS: TestAccXRaySamplingRule_tags (141.72s)
--- PASS: TestAccXRayGroup_tags (141.77s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_overlapping (95.24s)
--- PASS: TestAccXRayGroup_tags_DefaultTags_providerOnly (104.72s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Is there any benefit to generating actual Terraform source files rather than strings -- see https://github.com/hashicorp/terraform-provider-aws/pull/36719?

**Comment by gdavison** [comment]
> … perhaps we can add a page to the contributor docs outlining how to generate the tagging tests?

Definitely. There are a few more cases that I think we should address first

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

My only other thought was whether we should consider gating these tests (or at least adding an option to gate them for parent resources with long create times). The basic `_tags` test could always omit this so that it runs anytime the full resource suite runs, but gating the rest unless explicitly opted in might be a nice quality of life feature for long running tests. 

This is is no way a blocker, just an observation from the volume of tests touched here.

File: internal/service/accessanalyzer/analyzer_tags_gen_test.go [hunk 1]
```diff
@@ -40,24 +41,44 @@ func testAccAccessAnalyzerAnalyzer_tags(t *testing.T) {
...
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAnalyzerConfig_tags1(rName, "key1", "value1"),
+				ConfigDirectory: config.StaticDirectory("testdata/Analyzer/tags/"),
+				ConfigVariables: config.Variables{
+					"rName": config.StringVariable(rName),
+					"tags": config.MapVariable(map[string]config.Variable{
+						"key1": config.StringVariable("value1"),
+					}),
+				},
```
[the remark this answers is not in the document]
Comment (gdavison) [reply internal/service/accessanalyzer/analyzer_tags_gen_test.go:54]: There's a lot more code going on, especially for the `ImportStateVerify` steps, so I've been thinking of adding a comment to each step describing what it's doing

**Comment by gdavison** [comment]
> My only other thought was whether we should consider gating these tests (or at least adding an option to gate them for parent resources with long create times). The basic `_tags` test could always omit this so that it runs anytime the full resource suite runs, but gating the rest unless explicitly opted in might be a nice quality of life feature for long running tests.

Good point. I've created #37231

**Pull request merged**